sechub-sereco/src/main/java/com/daimler/sechub/sereco/importer/SarifV1JSONImporter.java

```diff
@@ -51,93 +59,196 @@ public SerecoMetaData importResult(String data) throws IOException {
-    }
+    private void resolveTargetInformation(Rule rule, ResultData data, Run run) {
+        List<ReportingDescriptorRelationship> relationShips = rule.getRelationships();
+        for (ReportingDescriptorRelationship relationShip : relationShips) {
+            ReportingDescriptorReference target = relationShip.getTarget();
+            if (target == null) {
+                continue;
+            }
+            String id = target.getId();
+            if (id == null) {
+                continue;
+            }
+            ToolComponentReference toolComponent = target.getToolComponent();
+            if (toolComponent == null) {
+                continue;
+            }
+            String toolComponentName = toolComponent.getName();
+            if (toolComponentName == null) {
+                continue;
+            }
 
-    private String resolveShortName(ReportingDescriptor desc) {
-        String name = desc == null ? null : desc.getName();
-        int index = name.lastIndexOf("/");
-        if (index != -1) {
-            index++;
-            if (index < name.length() - 1) {
-                name = name.substring(index);
+            if (CWE.equalsIgnoreCase(toolComponentName)) {
+                /* cwe found, so lets look after the id */
+                data.cweId = id;
             }
         }
-        return name;
     }
 
-    private SarifSchema210 readSarifSchema210(String data) throws JsonProcessingException, JsonMappingException {
-        return mapper.readValue(data, SarifSchema210.class);
+    private String resolveDescription(Rule rule) {
+        return rule.getFullDescription().getText();
+    }
+
+    private String resolveType(Rule rule, Run run) {
+        if (rule == null) {
+            return "error:rule==null!";
+        }
+        String type = null;
+        Message shortDescription = rule.getShortDescription();
+        if (shortDescription != null) {
+            type = shortDescription.getText();
+        }
+
+        if (type == null) {
+            type = handleSpecialToolTypeResolving(rule, run);
+        }
+
+        if (type == null) {
+            /*
+             * no type identifier found, so do fallback to id, we do not use "name" becaus
+             * this is for i18n!
+             */
+            type = rule.getId();
+        }
+        return type;
+    }
+
+    private String handleSpecialToolTypeResolving(Rule rule, Run run) {
+        Tool tool = run.getTool();
+        if (tool == null) {
+            return null;
+        }
+        Driver driver = tool.getDriver();
+        if (driver == null) {
+            return null;
+        }
+        String toolName = driver.getName();
+        if (toolName == null) {
+            return null;
+        }
+        if (toolName.equalsIgnoreCase("Brakeman")) {
+            return handleSpecialBrakemanResolving(rule, run);
+        }
+        return null;
+    }
+
+    private String handleSpecialBrakemanResolving(Rule rule, Run run) {
```

Comment:
I am not in favor of treating Brakeman any different than any other tool. Please remove this.

Reply:
already removed - see former messages.